### PR TITLE
update project member api for proj assignment screen

### PR DESCRIPTION
**Pull request by emmaacha**
Ticket ID: [Create Project Assignments Chart Screen#274](https://github.com/nerubia/kh360-react/issues/274)

FE PR: [add project assignments chart screen #367](https://github.com/nerubia/kh360-react/pull/367)

### CODE REVIEW

File: src/controllers/admin/project-member-controller.ts [hunk 1]
```diff
@@ -17,12 +17,14 @@ import { createProjectMemberSchema } from "../../utils/validation/project-member
 export const search = async (req: Request, res: Response) => {
   try {
     const { start_date, end_date, name, project_name, role, user_id, overlap } = req.query
+    const projectRole = role === "all" ? "" : role
```
Comment (jatnerubia) [inline src/controllers/admin/project-member-controller.ts:20]: handle this in service

File: src/services/project-member-service.ts [hunk 2]
```diff
@@ -107,40 +112,41 @@ export const getAllByFilters = async (
...
+        project_id: null,
+      })
+    }
   }
 
   if (role !== undefined) {
-    Object.assign(filter, {
-      project_role_id: parseInt(role),
-    })
+    if (role.trim().length > 0) {
```
Comment (jatnerubia) [inline src/services/project-member-service.ts:145]: do we need trim here?

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

File: src/services/project-member-service.ts (hunk 2 above)
Comment (emmaacha) [reply src/services/project-member-service.ts:145]: Ah, will be changing the empty string for this line `const projectRole = role === "all" ? "" : role` to undefined instead. Thanks!

**Comment by emmaacha** [comment]
Done ^^

**Review by jatnerubia — APPROVED** [review]
Cool

**Review by Dev-Nino — APPROVED** [review]

**Pull request merged**